cmake/BundledRocksDB.cmake

```diff
@@ -63,10 +64,17 @@ function(use_bundled_rocksdb SOURCE_DIR BINARY_DIR)
                 -DROCKSDB_INSTALL_ON_WINDOWS=ON
                 -DWITH_XPRESS=ON)
     else()
+    if(MINIFI_ZSTD_SOURCE STREQUAL "CONAN" AND MINIFI_ZLIB_SOURCE STREQUAL "CONAN")
+        list(APPEND ROCKSDB_CMAKE_ARGS
+            -DWITH_ZLIB=OFF
+            -DWITH_ZSTD=OFF)
+    else()
```

Comment:
I think we should fix RocksDB with Conan zlib and zstd instead, if that's possible. Or is that not possible or feasible?

Reply:
It is possible to fix RocksDB with conan zlib and zstd. I will try to add support for RocksDB conan as alternative to RocksDB cmake, so we can undo the updates that I made to BundledRocksDB.cmake, so that BundledRocksDB.cmake just stays standalone CMake specific and separate from conan. I will follow up on this.

Reply:
@szaszm @lordgamez 
I just added a **RocksDB** conan recipe, so we can create that conan package and build it with the same patches that we use to build RocksDB using the BundledRocksDB approach. However, now **RocksDB** conan package is separate from RocksDB **BundledRocksDB**, which uses standalone CMake. 

- I will make sure to update the conan steps in the PR's description from earlier. 
- I already added this update into the **CONAN.md** by adding an appendix section that shows how to create a custom conan package, which in our case is **RocksDB** due to needing to apply patches during the conan package creation.

With **RocksDB** as a conan package, it uses **zlib** and **zstd** conan packages when creating the **RocksDB** conan package.

Thus, I switched **BundledRocksDB** back to its original self based on **main** branch before I added those conditional statements that accounted for **Conan**.

Now when we use **RocksDB**, we run **get_rocksdb()** to retrieve **RocksDB** using **Conan** or **standalone CMake** depending on how the build is configured.

After this update, I rebuilt **MiNiFi** using conan, reran the CTESTs **218** out of **219** passed.
I rebuilt **MiNiFi** using standalone CMake, reran the CTESTs **218** out of **219** passed.

Reply:
Here are the conan steps for quick reference @szaszm @lordgamez :

~~~bash
sudo pip install --force-reinstall -v "conan==2.0.17"

# conanfile.py is in root dir of MiNiFi C++ project
cd $HOME/nifi-minifi-cpp

# create MINIFI_HOME env variable for binary executable minifi
export MINIFI_HOME=$(pwd)

# In case you need to delete all conan packages
# conan remove "*" -c

# Create custom RocksDB conan package with MiNiFi's patches applied
pushd thirdparty/rocksdb/all

conan create . --user=minifi --channel=develop --version=8.10.2 -pr=../../../etc/conan/profiles/release-linux

popd # Exit thirdparty/rocksdb/all

# install conan packages for MiNiFi C++ using conanfile.py invoking Conan
# since we created default profile earlier, we can override it with our own minifi profile
# make sure path is correct; Our conanfile.py also includes our custom RocksDB conan package
conan install . --build=missing --output-folder=build_conan -pr=$HOME/nifi-minifi-cpp/etc/conan/profiles/release-linux

# build MiNiFi C++ using conanfile.py invoking Conan & CMake
conan build . --output-folder=build_conan -pr=$HOME/nifi-minifi-cpp/etc/conan/profiles/release-linux
~~~